search-app/src/cmr/search/services/content_service.clj

```diff
@@ -0,0 +1,150 @@
+  []
+  (stl-cache/create-single-thread-lookup-cache
+   (cubby-cache/create-cubby-cache)))
+
+(defn get-page-content
+  "Given a context and route (optionally provider id and tag), get the
+  output of the page template that corresponds to the route."
+  ([context route]
+   (case route
+     "/sitemap.xml" (pages/sitemap-master context)
+     "/site/sitemap.xml" (pages/sitemap-top-level context)
+     "/site/collections/directory/eosdis" (pages/eosdis-collections-directory context)))
+  ([context route provider-id tag]
+   (cond
+     (string/ends-with? route ".xml")
+     (pages/sitemap-provider-tag context provider-id tag)
+     :else (pages/provider-tag-directory context provider-id tag))))
+
+(defn cache-page
+  "Cache a page for a given route. If the content is supplied, simply cache
+  that; if only a route is supplied, get the content for that route and then
+  cache it. If a provider id and tag are supplied, get the appopriate content
+  given those args, and cache the result."
+  ([context route]
+   (cache-page context route (get-page-content context route)))
+  ([context route content]
+   (debug "Caching" route)
+   (cache/set-value (cache/context->cache context cache-key) route content))
+  ([context route provider-id tag]
+    (cache-page context
+                route
+                (get-page-content context route provider-id tag))))
+
+(defn- create-lookup-fn
+  [args]
+  (debug "Route" (second args) "not found in cache")
+  (fn [] (apply get-page-content args)))
+
+(defn retrieve-page
+  "Attempt to retrieve from the cache the page data for a given route. If no
+  value is found in the cache, genereate the content, cache it, and then return
+  the cached value.
+
+  Note that the first three arguements will always be `context`, HTTP `params`,
+  and resource `route`. Additional arguments may be present (e.g., provider id
+  and tag)."
+  [context params route & remaining]
+  (debug "Retrieving" route "with params" params "and remaining args" remaining)
+  (let [args (concat [context route] remaining)
+        regenerate? (= "true" (util/safe-lowercase (:regenerate params)))]
+    ;; Only admins can force the pages and sitemaps in the cache to be
+    ;; regenerated.
+    (when regenerate?
+      (acl/verify-ingest-management-permission context :update)
+      (debug "Forcing resource regeneration/cache update" route)
+      (apply cache-page args))
+    (cache/get-value (cache/context->cache context cache-key)
+                     route
+                     (create-lookup-fn args))))
+
+(defn- cache-top-level-content
+  "Cache all the content for the expensive, top-level routes."
+  [context]
+  (cache-page context "/sitemap.xml")
+  (cache-page context "/site/sitemap.xml")
+  (cache-page context "/site/collections/directory/eosdis"))
+
+(defn- cache-directory-content
+  "Cache all the content for the expensive, directory-level routes."
+  [context]
+  (debug "Providers:" (mdb/get-providers context))
```

Comment:
Remove these two debug lines

Reply:
This is logging ... this is the sort of info we'd want to see when setting the log level to debug. Why do you want to remove it?

Reply:
I've updated the debug logging here to just use the display of provider-ids (no need for both).